src/main/java/ContinuousIntegrationServer.java

```diff
@@ -1,44 +1,179 @@
+            deleteDirectory(clonedRepoFile);
+        }
+        // Cloning the repository
+        try {
+            System.out.println("Cloning repository...");
+            Git.cloneRepository()
+                    .setURI(cloneUrl)
+                    .setDirectory(clonedRepoFile)
+                    .setBranch(branchName)
+                    .call().close();
+            System.out.println("Repository cloned successfully");
+        } catch (GitAPIException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Method that compiles the repository using maven commands, specifically "clean
+     * install"
+     *
+     * @param cloneddirectoryPath - Path to the cloned repository
+     *                            (src/main/resources/)
+     * @param clonedRepoFile      - File object representing the cloned repository
+     */
+    public void compileRepository(String cloneddirectoryPath, File clonedRepoFile) {
+        InvocationRequest invocationRequest = new DefaultInvocationRequest();
+        invocationRequest.setPomFile(new File(cloneddirectoryPath, "pom.xml")); // pom.xml is the file that contains the
+        // maven configuration
+        invocationRequest.setBaseDirectory(clonedRepoFile);
+        invocationRequest.setGoals(Collections.singletonList("clean install"));
+
+        DefaultInvoker invoker = new DefaultInvoker(); // Invoker is used to execute the maven commands
+
+        // Excuting the maven commands
+        try {
+            InvocationResult result = invoker.execute(invocationRequest);
+
+            if (result.getExitCode() == 0) {
+                System.out.println("Build successful!");
+            } else {
+                System.out.println("Build failed. Exit code: " + result.getExitCode());
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Method that deletes the cloned repository after the CI job is done to avoid
+     * clutter
+     * The method works recursively by deleting all files and directories within the
+     * cloned repository
+     *
+     * @param directory - File object representing the cloned repository
+     */
+    public void deleteDirectory(File directory) {
+        Path directoryPath = directory.toPath();
+        try {
+            Files.walk(directoryPath)
+                    .map(Path::toFile)
+                    .sorted((o1, o2) -> -o1.compareTo(o2))
+                    .forEach(File::delete);
+            System.out.println("Local repository deleted successfully: " + directoryPath);
+        } catch (IOException e) {
+            System.err.println("Error deleting local repository: " + e.getMessage());
+        }
+    }
+
     // used to start the CI server in command line
     public static void main(String[] args) throws Exception {
-        Server server = new Server(8013);
+        Server server = new Server(8040);
```

Comment:
We had used a server port that we thought no one used. This was due to the server not working as intended since it was "already in use". Therefore, we changed it to _8040_ for testing. We could maybe change it back to _8013_ now. 

> 

```suggestion
        Server server = new Server(8013);
```